src/Bicep.LangServer.IntegrationTests/FileWatcherTests.cs

```diff
@@ -205,37 +209,107 @@ public async Task Background_folder_deletion_should_trigger_a_recompilation()
-                services => services.WithFileResolver(fileResolver));
-            var client = helper.Client;
-
-            // open the main document
-            {
-                client.TextDocument.DidOpenTextDocument(TextDocumentParamHelper.CreateDidOpenDocumentParams(mainUri, fileSystemDict[mainUri.ToUriEncoded()], 1));
-
-                var diagsParams = await diagsListener.WaitNext();
-                diagsParams.Uri.Should().Be(mainUri);
-                diagsParams.Diagnostics.Should().Contain(x => x.Message.Contains("The specified \"object\" declaration is missing the following required properties: \"requiredIpnut\"."));
-            }
-
-            // delete the module folder with a background process
-            {
-                fileResolver.MockFileSystem.File.Delete(moduleUri.GetFileSystemPath());
-                SendDidChangeWatchedFiles(client, (DocumentUri.FromFileSystemPath("/path/toOther"), FileChangeType.Deleted));
-
-                var nextDiags = await diagsListener.WaitNext();
-                nextDiags.Uri.Should().Be(mainUri);
-                nextDiags.Diagnostics.Should().Contain(x => x.Message.Contains("An error occurred reading file. Could not find file '/path/toOther/module.bicep'."));
-            }
+        var fileResolver = new InMemoryFileResolver(fileSystemDict);
+        using var helper = await LanguageServerHelper.StartServer(
+            this.TestContext,
+            options => options.OnPublishDiagnostics(diagsListener.AddMessage),
+            services => services.WithFileResolver(fileResolver));
+        var client = helper.Client;
+
+        // open the main document
+        {
+            client.TextDocument.DidOpenTextDocument(TextDocumentParamHelper.CreateDidOpenDocumentParams(mainUri, fileSystemDict[mainUri.ToUriEncoded()], 1));
+
+            var diagsParams = await diagsListener.WaitNext();
+            diagsParams.Uri.Should().Be(mainUri);
+            diagsParams.Diagnostics.Should().Contain(x => x.Message.Contains("The specified \"object\" declaration is missing the following required properties: \"requiredIpnut\"."));
+        }
+
+        // delete the module folder with a background process
+        {
+            fileResolver.MockFileSystem.File.Delete(moduleUri.GetFileSystemPath());
+            SendDidChangeWatchedFiles(client, (DocumentUri.FromFileSystemPath("/path/toOther"), FileChangeType.Deleted));
+
+            var nextDiags = await diagsListener.WaitNext();
+            nextDiags.Uri.Should().Be(mainUri);
+            nextDiags.Diagnostics.Should().Contain(x => x.Message.Contains("An error occurred reading file. Could not find file '/path/toOther/module.bicep'."));
         }
     }
+
+    [TestMethod]
+    public async Task Auxiliary_file_access_is_cached()
+    {
+        var bicepUri = new Uri("file:///main.bicep");
+        var bicepContents = """
+#disable-next-line no-unused-params
+param foo 'abc' = loadTextContent('path/to/bar.txt')
+""";
```

Comment:
[nit]: margings are marked by the last triple quotes in this statement, having it all the way to the left makes my OCD trigger :D, maybe you can keep it indented so its easier to read?